### DISCUSSION TITLE
New ringlight design thoughts: digital potentiometers and light blending

### DISCUSSION THREAD

**Discussion by sokol07** [OpenScanEu/OpenScan-PCB#3] [Ideas]
Hi!
I've got two thoughts about the new ringlight design.
First of all, speaking of dimming the LEDs and non-PWM solution - I believe the answer is quite simple, we need to implement a digital potentiometer there, controlled by the RPi/Ardu. The selection is a matter of necessary resistance (so dependable on the voltage and LEDs selection as we need to adjust the current) and interface (3-wire, I2C, SPI, etc).
My second thought is about the light blending - not long ago I was developing and setting up an industrial machine with vision system for product positioning. The product was mostly glass bottles so the camera needed a decent light source. The ringlights placed around the camera were also LEDs but the key point is that the ringlights had milk-glass-like blending screens, made from polycarbonate or other plastic. The LEDs light was a little weaker due to the blends but it was also more uniform and smooth, as the light sources were not reflecting as points in the objects. Maybe it would be a benefit to add such a blending screen, made of a milky plastic, in front of the LEDs in the new ringlight design?

**Comment by sokol07** [comment]
@OpenScanEu bump of the discussion :)

**Comment by OpenScanEu** [comment]
@sokol07 I just got back to the LED-topic and I am not 100% convinced that you really need dimming, since you can adjust the shutter value. Or do I miss something? (But for sure, your approach should definitely work)

Concerning the lightblending: I originally designed a diffusor plate, which can be printed in translucent Filament and is placed in front of the LEDs. This will make the lightsource quite a bit softer. The problem with the original design was, that there is a lot of back scattering, so that the photo gets a white-ish fog effect (sorry, lacking words here). I started redesigning this module and added a black/non-translucent part around the camera, which solved this issue. BUT I did not manage to document/publish the design files yet:
![69Q4b8ozZeIqOW09](https://user-images.githubusercontent.com/57842400/169020299-55aad1cb-2bf1-4d0a-9d61-dc485cbad4d7.jpg)

(Not sure, whether you got this one?)

**Comment by sokol07** [comment]
The only situation when dimming could be useful is a very small object with complicated shape - you have to move the camera close to the object because it is small and turn on all LEDs to reduce shadows. I had such a case (IIRC, I was scanning something of a size of a small safety pin) and the fastest shutter still had some problems with overxposure because I was really close to the object.

The diffuser plate in the photos look nice, I guess it should do its job. The hood around the lens is definitely good idea. My only concern is the fact that you need translucent filament to print this - I haven't ever tried such filament and I don't have one. I guess I could find one but it's the questions of quality and transparency - I guess some of the filaments will be "transparent" but not enough and they will be dimming the light. I was thinking if we could prepare model of a diffuser with the main part made of acrylic sheet - you can find these milky sheets for example on amazon:
https://www.amazon.com/Milky-Acrylic-Translucent-Plexiglass-AZM/dp/B0818HYY8G/ref=sr_1_2
However, if the transparent filament works ok, there are no transparency issue and problems with printing it - it is ok too!

**Reply by OpenScanEu** [reply]
The filament I use is not really transparent, it is called PETG natural (or alternatively PLA natural) and it is translucent enough. The wide range of shutter speed leaves enough room for adjustments.
Concerning the very fast shutter speeds, you can even go below 1ms (left limit of the slider), by adjusting the slider's range in the flow:
![image](https://user-images.githubusercontent.com/57842400/169341896-9d25e736-d8bd-4e49-9d10-fef7930ab07d.png)
I have done a quick test and values below 0.1 don't seem to have any effect on the exposure.

![image](https://user-images.githubusercontent.com/57842400/169342071-d8eeb418-8c18-4b76-bbe0-a5b53f35973d.png)
left is done with 1ms and right with 0.1ms (but 0.01ms looks the same, so I think this is a hardware/driver limitation somewhere in that range)

**Comment by sokol07** [comment]
You're right, the wide range of shutter times is enough in 99% of situations I guess. The dimming maybe could be useful at some point in future if we, for example, switch to more powerful LEDs or build a totally different device (bigger? smaller?) but controlled with the same board - in this case the dimming could be useful. However, I think that the dimming feature is less crucial at the moment than the diffusor.
